From 963124b72acc77399500f0c74e88ee65ea00d2be Mon Sep 17 00:00:00 2001
From: "Bhakte, GurudattaX" <gurudattax.bhakte@intel.com>
Date: Mon, 1 Jun 2015 15:40:03 +0530
Subject: [PATCH] ASoC: Intel: Fix for KW critical error in kernel code.

Issue: GMINL-10371

The variable 'cmd.cfg.rate' might be used uninitialized.
This patch improves the code to avoid the usage of variable 'cmd.cfg.rate'
uninitialized.

Change-Id: If2e970a7edec68d648d8639d6d17cf60aa8e8779
Signed-off-by: Bhakte, GurudattaX <gurudattax.bhakte@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 5ab9337129cf..e67079835305 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -1316,6 +1316,9 @@ static int sst_send_speech_path(struct sst_data *sst, u16 switch_state)
 		cmd.cfg.rate = 0;
 		cmd.cfg.bwx = 1;
 		break;
+	default:
+		cmd.cfg.rate = 0;
+		break;
 	}
 	return sst_fill_and_send_cmd(sst, SST_IPC_IA_CMD, SST_FLAG_BLOCKED,
 				     SST_TASK_SBA, 0, &cmd,
-- 
1.9.1

